src/shorah/amplicon.py

```diff
@@ -325,8 +326,8 @@ def main(args):
     ref_seq = list(SeqIO.parse(in_fasta, 'fasta'))[0]
     ref_name = ref_seq.id
```

Comment:
Would it be easier and also add some robustness to just clean the reference id from special characters?  Something like `ref_name = re.sub(r'[^\w\s]', '_', ref_seq.id)`?

Reply:
I was thinking about something like that too. (And going even further : like `[^[:alnum:]_]` or `[^\w]` with ASCII rule flag).

The main reasons I am hesitating :

 - Change of file names compared to legacy data.

   (shorah 2.0 would end up using different names than shorah 1.x. Scripts relying on these will need to be changed and shorah 2.0 will not necessarily recognize files processed with older versions)

 - the gcc version (4.8) used by bioconda is an older one and doesn't provide the necessary `libstdcxx` for the c++11 regex support (it appears in gcc 4.9+).

   That would either require to use boost::regex as a fall back  on bioconda.

   Or working around using `algorithm`'s `replace_if` and `isalnum`

On the other hand, a major version bump like *1.x* => *2.0* would be a good opportunity to introduce breaking changes.

Reply:
Scripts will have to be adapted anyway.

I'm not really sure I'm getting the point about gcc, it's long time since I wrote it. Since we are sanitising sequence and, hence, file names in python, isn't it enough to find the last (and only) dot with `rfind`? Where is the issue?

Reply:
File names (`w-%s-%u-%u.read.fas`) as also generated [in the C++ side of stuff](https://github.com/cbg-ethz/shorah/blob/b17422e0f45b3ee53bc5882d0ca2baa3e506ba7f/src/cpp/b2w.cpp#L246).
So we should have the exact same file name sanitation on both Python and C++.
Which is problematic because the dinosaur of a GCC version (4.8) still used by bioconda lacks c++11 regex support so we won't be able to use the exact same `re.sub` command on both sides.